packages/core/src/routes/profile/index.ts

```diff
@@ -76,7 +79,14 @@ export default function profileRoutes<T extends UserRouter>(
       const { id: userId } = ctx.auth;
       const { password, verificationRecordId } = ctx.guard.body;
 
-      // TODO(LOG-9947): apply password policy
+      const user = await findUserById(userId);
+      const signInExperience = await findDefaultSignInExperience();
+      const passwordPolicyChecker = new PasswordPolicyChecker(signInExperience.passwordPolicy);
+      const issues = await checkPasswordPolicyForUser(passwordPolicyChecker, password, user);
+
+      if (issues.length > 0) {
+        throw new RequestError('password.rejected', { issues });
+      }
```

Comment:
Also, the new password can not be the same as the existing one.  P.S Can reuse the `PasswordValidator` under experience routes.